File: CommonConfig.java
import java.io.*;

public class CommonConfig {
    int numberOfPreferredNeighbors;
    int unchokingInterval;
    int optimisticUnchokingInterval;
    String fileName;
    int fileSize;
    int pieceSize;

    public int getNumberOfPieces() {
        return (int) Math.ceil((double) fileSize / pieceSize);
    }

    public static CommonConfig load(String path) {
        CommonConfig config = new CommonConfig();
        try (BufferedReader br = new BufferedReader(new FileReader(path))) {
            String line;
            while ((line = br.readLine()) != null) {
                line = line.trim();
                if (line.isEmpty()) continue;
                String[] parts = line.split("\\s+");
                switch (parts[0]) {
                    case "NumberOfPreferredNeighbors":
                        config.numberOfPreferredNeighbors = Integer.parseInt(parts[1]);
                        break;
                    case "UnchokingInterval":
                        config.unchokingInterval = Integer.parseInt(parts[1]);
                        break;
                    case "OptimisticUnchokingInterval":
                        config.optimisticUnchokingInterval = Integer.parseInt(parts[1]);
                        break;
                    case "FileName":
                        config.fileName = parts[1];
                        break;
                    case "FileSize":
                        config.fileSize = Integer.parseInt(parts[1]);
                        break;
                    case "PieceSize":
                        config.pieceSize = Integer.parseInt(parts[1]);
                        break;
                }
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
        return config;
    }
}

File: common_config.py
import math


class CommonConfig:
    def __init__(self):
        self.number_of_preferred_neighbors = 0
        self.unchoking_interval = 0
        self.optimistic_unchoking_interval = 0
        self.file_name = ""
        self.file_size = 0
        self.piece_size = 0

    def get_number_of_pieces(self):
        return math.ceil(self.file_size / self.piece_size)

    @staticmethod
    def load(path):
        config = CommonConfig()
        with open(path, "r") as f:
            for line in f:
                line = line.strip()
                if not line:
                    continue
                parts = line.split()
                key, value = parts[0], parts[1]
                if key == "NumberOfPreferredNeighbors":
                    config.number_of_preferred_neighbors = int(value)
                elif key == "UnchokingInterval":
                    config.unchoking_interval = int(value)
                elif key == "OptimisticUnchokingInterval":
                    config.optimistic_unchoking_interval = int(value)
                elif key == "FileName":
                    config.file_name = value
                elif key == "FileSize":
                    config.file_size = int(value)
                elif key == "PieceSize":
                    config.piece_size = int(value)
        return config
